brocade-switches.yaml

```diff
@@ -0,0 +1,109 @@
+- name: "Gather required information"
+  hosts: brocade
+  gather_facts: no
+  tasks:
+
+    # it's important to get the rbridge id to get the hostname
+    - name: "Get rbridge ID"
+      community.network.nos_command:
+        commands: >-
+          show chassis | inc Rbridge
+      register: rbridge_stdout
+      vars:
+        ansible_command_timeout: 90
+
+    - name: "Get hostname"
+      community.network.nos_command:
+        commands: >-
+          show running-config rbridge-id "{{rbridge_id}}" | inc host-name
+      register: hostname_stdout
+      vars:
+        ansible_command_timeout: 90
+        rbridge_id: "{{ rbridge_stdout.stdout[0].split(':')[1] }}"
+        # need to extract the id from an output like this
+        # "stdout": ["Rbridge-id: 4"]
+
+    - name: "Get switch model"
+      community.network.nos_command:
+        commands: >-
+          show chassis | inc Chassis
+      register: chassis_stdout
+      vars:
+        ansible_command_timeout: 90
+
+    - name: "Get serial number"
+      community.network.nos_command:
+        commands: >-
+          show chassis | inc Serial | last 1
+      register: serial_stdout
+      vars:
+        ansible_command_timeout: 90
+
+    - name: "Calcuate rack and rackpos"
+      set_fact:
+        rack: "{{ parsed_address[-2] }}"
+        rackpos: "{{ parsed_address[-1] }}"
+      vars:
+        parsed_address: "{{ ansible_host | split('.') }}"
+
+    - name: "Set hostname"
+      set_fact:
+        brocade_hostname: "{{ hostname_stdout.stdout[0].split(' ')[2] }} "
+        brocade_serial: "{{ serial_stdout.stdout[0].split(':')[1] }}"
+        brocade_model: "{{ chassis_stdout.stdout[0].split(':')[1] }}"
+        rbridge_id: "{{ rbridge_stdout.stdout[0].split(':')[1] }}"
+
+    - name: "Create the device_type"
+      ignore_errors: true
```

Comment:
Why do we `ignore_errors` here?

Reply:
It was failing when we re-run the playbooks. It complains about creating the device_type again. 

Similarly it doesn't like when I create a device with a specific location in rack again. It complained about the rack position being occupied.

However, now I decided to remove it and it does not complain, instead every time I run the playbook it appears something about the device_type and device has changed so this task "updates" it. 
```
TASK [Create the device_type] ********************************************************************************************************************************************
task path: /Users/naved/office-stuff/moc-ansible-netbox/brocade-switches.yaml:55
changed: [brocade-3-43 -> 127.0.0.1] => {"changed": true, "device_type": {"comments": "", "created": "2022-02-23", "custom_fields": {}, "device_count": 1, "display": "BR-VDX6740", "front_image": null, "id": 49, "is_full_depth": true, "last_updated": "2022-02-24T14:44:57.110940Z", "manufacturer": 6, "model": "\tBR-VDX6740", "part_number": "", "rear_image": null, "slug": "-br-vdx6740", "subdevice_role": null, "tags": [], "u_height": 1, "url": "https://netbox.apps.ocp-prod.massopen.cloud/api/dcim/device-types/49/"}, "msg": "device_type \tBR-VDX6740 updated"}

TASK [Register the device in netbox] *************************************************************************************************************************************
task path: /Users/naved/office-stuff/moc-ansible-netbox/brocade-switches.yaml:65
changed: [brocade-3-43 -> 127.0.0.1] => {"changed": true, "device": {"asset_tag": null, "cluster": null, "comments": "", "config_context": {}, "created": "2022-02-24", "custom_fields": {}, "device_role": 40, "device_type": 49, "display": "rbr-04", "face": "front", "id": 83, "last_updated": "2022-02-24T14:44:58.485708Z", "local_context_data": null, "location": 36, "name": "rbr-04", "parent_device": null, "platform": null, "position": 43, "primary_ip": null, "primary_ip4": null, "primary_ip6": null, "rack": 40, "serial": "    \tCPL2547K0RS", "site": 3, "status": "active", "tags": [], "tenant": null, "url": "https://netbox.apps.ocp-prod.massopen.cloud/api/dcim/devices/83/", "vc_position": null, "vc_priority": null, "virtual_chassis": null}, "msg": "device rbr-04 updated"}
```

If you look into netbox, it documents the changelog and what got updated. Looks like the only thing that got changes is `last_updated`...
https://netbox.apps.ocp-prod.massopen.cloud/extras/changelog/1178/

But looks like I don't need to `ignore_errors` anymore.